tensorboard/backend/auth_context_test.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for `tensorboard.backend.auth_context`."""
+from werkzeug import test as werkzeug_test
+from werkzeug import wrappers
+
+from tensorboard import auth
+from tensorboard import context
+from tensorboard import test as tb_test
+from tensorboard.backend import auth_context
+
+
+class SimpleAuthProvider(auth.AuthProvider):
+    """Simple AuthProvider that returns the value it is initialized with."""
+
+    def __init__(self, credential):
+        self._credential = credential
+
+    def authenticate(self, environ):
+        return self._credential
+
+
+def _create_auth_provider_verifier_app(expected_auth_key):
+    """Generates a WSGI application for verifying AuthContextMiddleware.
+
+    It should be placed downstream from the AuthContextMiddleware. It will
```

Comment:
I just tried to reword since adding an example just duplicates the code below. Since the WSGI PEP (https://peps.python.org/pep-0333/) makes a passing reference to the "Chain of Responsibility" pattern, I reworded in those terms.